monkey/infection_monkey/master/ip_scanner.py

```
@@ -61,16 +61,30 @@ def _scan_addresses(
                 address = addresses.get_nowait()
                 logger.info(f"Scanning {address.ip}")
 
-                # TODO: Catch exceptions to prevent thread from crashing
-                ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
-                port_scan_data = self._puppet.scan_tcp_ports(address.ip, tcp_ports, tcp_timeout)
+                try:
+                    ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
+                except Exception as ex:
```

Same questions. Why do we catch all exceptions? Why we log it on debug level?